x-pack/plugins/drilldowns/public/plugin.ts

```diff
@@ -6,52 +6,41 @@
 
 import { CoreStart, CoreSetup, Plugin } from 'src/core/public';
 import { UiActionsSetup, UiActionsStart } from '../../../../src/plugins/ui_actions/public';
-import { DrilldownService } from './service';
-import {
-  FlyoutCreateDrilldownActionContext,
-  FlyoutEditDrilldownActionContext,
-  OPEN_FLYOUT_ADD_DRILLDOWN,
-  OPEN_FLYOUT_EDIT_DRILLDOWN,
-} from './actions';
-
-export interface DrilldownsSetupDependencies {
+import { AdvancedUiActionsSetup, AdvancedUiActionsStart } from '../../advanced_ui_actions/public';
+import { createFlyoutManageDrilldowns } from './components/connected_flyout_manage_drilldowns';
+import { Storage } from '../../../../src/plugins/kibana_utils/public';
+
+export interface SetupDependencies {
   uiActions: UiActionsSetup;
+  advancedUiActions: AdvancedUiActionsSetup;
 }
 
-export interface DrilldownsStartDependencies {
+export interface StartDependencies {
   uiActions: UiActionsStart;
+  advancedUiActions: AdvancedUiActionsStart;
 }
 
-export type DrilldownsSetupContract = Pick<DrilldownService, 'registerDrilldown'>;
-
 // eslint-disable-next-line
-export interface DrilldownsStartContract {}
+export interface SetupContract {}
 
-declare module '../../../../src/plugins/ui_actions/public' {
-  export interface ActionContextMapping {
-    [OPEN_FLYOUT_ADD_DRILLDOWN]: FlyoutCreateDrilldownActionContext;
-    [OPEN_FLYOUT_EDIT_DRILLDOWN]: FlyoutEditDrilldownActionContext;
-  }
+export interface StartContract {
+  FlyoutManageDrilldowns: ReturnType<typeof createFlyoutManageDrilldowns>;
 }
 
 export class DrilldownsPlugin
-  implements
-    Plugin<
-      DrilldownsSetupContract,
-      DrilldownsStartContract,
-      DrilldownsSetupDependencies,
-      DrilldownsStartDependencies
-    > {
-  private readonly service = new DrilldownService();
-
-  public setup(core: CoreSetup, plugins: DrilldownsSetupDependencies): DrilldownsSetupContract {
-    this.service.bootstrap(core, plugins);
-
-    return this.service;
+  implements Plugin<SetupContract, StartContract, SetupDependencies, StartDependencies> {
+  public setup(core: CoreSetup, plugins: SetupDependencies): SetupContract {
+    return {};
   }
 
-  public start(core: CoreStart, plugins: DrilldownsStartDependencies): DrilldownsStartContract {
-    return {};
+  public start(core: CoreStart, plugins: StartDependencies): StartContract {
+    return {
```

Comment:
Why not merge with ui_actions_enhanced and remove this plugin entirely? Since the drilldown registration is in that plugin, i'd expect to find this there too (and then you can get rid of this whole plugin).

Reply:
> Why not merge with ui_actions_enhanced and remove this plugin entirely?

That is the plan, we have a task for that scheduled: https://github.com/elastic/kibana/issues/59562

Reply:
I know this PR has gone through a lot of iterations, and everyone is eager to see it get merged, but this is introducing a new public API, in a brand new plugin, knowing that we are immediately going to remove it in a follow up.  It may be out there for other developers to use and build on top of if the follow up doesn't get into 7.8, and we have to assume there is a chance it won't.

We don't _yet_ promise developers we won't make breaking API changes in minors, but we do want to get there eventually.  Good to start practicing now, unless good reason not to.  Why not move in this PR so we don't expose an API that we already know we will make breaking changes to?

Reply:
At our sync agreed to move it into `ui_actions_enhanced` plugin as planned in subsequent task.